scripts/kconfig/kconfig.py

```diff
@@ -173,32 +176,38 @@ def promptless(sym):
     return not any(node.prompt for node in sym.nodes)
 
 
-def write_kconfig_filenames(paths, root_path, output_file_path):
-    # 'paths' is a list of paths. The list has duplicates and the
-    # paths are either absolute or relative to 'root_path'.
+def write_kconfig_filenames(kconf, kconfig_list_path):
+    # Writes a sorted list with the absolute paths of all parsed Kconfig files
+    # to 'kconfig_list_path'. The paths are realpath()'d, and duplicates are
+    # removed. This file is used by CMake to look for changed Kconfig files. It
+    # needs to be deterministic.
 
-    # We need to write this list, in a format that CMake can easily
-    # parse, to the output file at 'output_file_path'.
-
-    # The written list has sorted real (absolute) paths, and it does not have
-    # duplicates. The list is sorted to be deterministic. It is realpath()'d
-    # to ensure that different representations of the same path does not end
-    # up with two entries, as that could cause the build system to fail.
-
-    with open(output_file_path, 'w') as out:
-        for path in sorted({os.path.realpath(os.path.join(root_path, path))
-                            for path in paths}):
+    with open(kconfig_list_path, 'w') as out:
+        for path in sorted({os.path.realpath(os.path.join(kconf.srctree, path))
+                            for path in kconf.kconfig_filenames}):
             print(path, file=out)
 
 
 def parse_args():
     parser = argparse.ArgumentParser()
 
-    parser.add_argument("kconfig_root")
-    parser.add_argument("dotconfig")
-    parser.add_argument("autoconf")
-    parser.add_argument("sources")
-    parser.add_argument("conf_fragments", nargs='+')
+    parser.add_argument("--handwritten-input-configs",
+                        action="store_true",
+                        help="Assume the input configuration files are "
+                             "handwritten fragments and do additional checks "
+                             "on them, like no promptless symbols being "
+                             "assigned")
+    parser.add_argument("kconfig",
+                        help="Top-level Kconfig file")
+    parser.add_argument("config_out",
+                        help="Output configuration file")
```

Comment:
Configuration file is too ambiguous, let's either use the term "Kconfig fragment" or "fragment" to clarify.

Using _out and _in suffixes makes sense.

Reply:
The output configuration file is a not a fragment, so probably shouldn't have "fragment" in it.

Reply:
ah, right, I consider the

A=B syntax to be

of the filetype "Kconfig fragment",

but I guess technically fragment also implies that it's just a subset of options.

Then I don't know how to make this clearer, so this will have to do.